build/lint.mk

```diff
@@ -0,0 +1,38 @@
+################################################################################
+###                             Required Variables                           ###
+################################################################################
+ifndef DOCKER
+$(error DOCKER not set)
+endif
+
+ifndef BUILD_CACHE_DIR
+$(error BUILD_CACHE_DIR not set)
+endif
+
+################################################################################
+###                             Lint Settings                                ###
+################################################################################
+
+LINT_FROM_REV ?= $(shell git merge-base origin/master HEAD)
+
+GOLANGCI_VERSION ?= $(shell cat .golangci-version)
+GOLANGCI_IMAGE_TAG ?= golangci/golangci-lint:$(GOLANGCI_VERSION)
+
+GOLANGCI_CACHE_DIR ?= $(CURDIR)/$(BUILD_CACHE_DIR)/golangci-lint/$(GOLANGCI_VERSION)
+
+################################################################################
+###                             Lint Target                                  ###
+################################################################################
+
+.PHONY: lint
+lint: $(GOLANGCI_CACHE_DIR)
+	@echo "Running lint from rev $(LINT_FROM_REV), use LINT_FROM_REV var to override."
+	$(DOCKER) run -t --rm \
+		-v $(GOLANGCI_CACHE_DIR):/root/cache \
+		-v $(CURDIR):/app \
+		-w /app \
+		$(GOLANGCI_IMAGE_TAG) \
+		golangci-lint run -v --timeout 10m --new-from-rev $(LINT_FROM_REV)
```

Comment:
`timeout` is already defined in `.golangci.yml`

Curious why are we using this, when there is also [`act`](https://github.com/nektos/act)